doc/source/ray-core/walkthrough.rst

```diff
@@ -2,125 +2,135 @@
-                // the `-Dray.address=<cluster-address>` java system property.
-                Ray.init();
-                ...
-            }
-        }
-
-.. tabbed:: C++
-
-    .. code-block:: c++
-
-        // Run `ray cpp --show-library-path` to find headers and libraries.
-        #include <ray/api.h>
-
-        int main(int argc, char **argv) {
-            // Start Ray runtime. If you're connecting to an existing cluster, you can set
-            // the `RAY_ADDRESS` env var.
-            ray::Init();
-            ...
-        }
-
-
-Ray will then be able to utilize all cores of your machine. Find out how to configure the number of cores Ray will use at :ref:`configuring-ray`.
-
-To start a multi-node Ray cluster, see the :ref:`cluster setup page <cluster-index>`.
-
-Using Tasks, Actors, and Objects
---------------------------------
 
-Click through below to walk through using Ray's key concepts: Tasks, Actors, and Objects.
+Defining the Progress Actor
+---------------------------
+Next, we will define a Ray actor that can be called by sampling tasks to update progress.
+Ray actors are essentially stateful services that anyone with an instance (a handle) of the actor can call its methods.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __defining_actor_start__
+    :end-before: __defining_actor_end__
+
+We define a Ray actor by decorating a normal Python class with :ref:`ray.remote() <ray-remote-ref>`.
+The progress actor has ``report_progress()`` method that will be called by sampling tasks to update their progress individually
+and ``get_progress()`` method to get the overall progress.
+
+Defining the Sampling Task
+--------------------------
+Next, we will define a Ray task that does the sampling up to the given times and returns the number of samples that are inside the circle.
+Ray tasks are stateless functions that will be executed asynchronously and in parallel.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __defining_task_start__
+    :end-before: __defining_task_end__
+
+We define a Ray task by decorating a normal Python function with :ref:`ray.remote() <ray-remote-ref>`.
+The sampling task takes a progress actor handle as an input and reports progress to it.
+This shows an example of calling actors from tasks.
+
+Creating a Progress Actor
+-------------------------
+Once the actor is defined, we can create an instance of it.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __creating_actor_start__
+    :end-before: __creating_actor_end__
+
+We create an instance of the progress actor by calling ``remote()`` method with arguments to the constructor.
+This will create and run the actor on a remote worker process.
+The return value of ``Actor.remote(...)`` is an actor handle that can be used to call the actor later on.
```

Comment:
The return value of ``Actor.remote(...)`` is an actor handle that can be used to call the actor methods.